src/set_theory/ordinal/cantor_normal_form.lean

```diff
@@ -37,14 +37,15 @@ open order
 namespace ordinal
 
 /-- Inducts on the base `b` expansion of an ordinal. -/
-@[elab_as_eliminator] noncomputable def CNF_rec {b : ordinal} (hb : b ≠ 0)
+@[elab_as_eliminator, nolint unused_arguments] noncomputable def CNF_rec {b : ordinal} (hb : b ≠ 0)
   {C : ordinal → Sort*} (H0 : C 0) (H : ∀ o, o ≠ 0 → C (o % b ^ log b o) → C o) : ∀ o, C o
 | o :=
   if ho : o = 0 then by rwa ho else
-  have _, from mod_opow_log_lt_self hb ho,
-  H o ho (CNF_rec (o % b ^ log b o))
+    let hwf := mod_opow_log_lt_self b ho in H o ho (CNF_rec (o % b ^ log b o))
 using_well_founded {dec_tac := `[assumption]}
 
+-- Todo (vihdzp): remove the unused hypothesis `b ≠ 0`.
```

Comment:
I think you should do that in this PR

Reply:
Which is to say: this pr should remove that argument but do nothing else; while the other PR should actually do the proposed design change.

Reply:
There goes.